pkg/client/cache/reflector.go

```diff
@@ -265,6 +265,7 @@ func (r *Reflector) ListAndWatch(stopCh <-chan struct{}) error {
 			switch err {
 			case io.EOF:
 				// watch closed normally
+				glog.V(1).Infof("%s: Watch for %v closed normally (with EOF)", r.name, r.expectedType)
```

Comment:
Hmmm... I think if we want to do a full relist periodically, we probably should rely on resyncPeriod rather than resyncPeriod.  I'm not sure what you mean by "we can't guarantee that the watch will be open always anyway".

I'm suggesting if this is a "normal" event, we should restart the watch (`continue` the loop).  But that would be a separate PR - this one is supposed to just add the log messages we need to take the next step.